aws-provider/postgresdb/README.md

```diff
@@ -0,0 +1,346 @@
+   >
+   > In the trust policy for this role, I am specifically mentioning all
+   > service accounts for the providers I require access to the account.
+   >
+   > Whilst this is a concious decision to provide visibility within the trust
+   > about which service accounts are allowed to connect, you may wish to
+   > allow any service account in a given namespace to connect as long as it
+   > starts with a provider prefix.
+   >
+   > In this instance, replace the entire `Condition` statement with an
+   > equivelant [`StringLike`](https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_elements_condition_operators.html#Conditions_String)
+   > condition.
+
+   The following example uses the `sponge` command from the `moreutils` package
+   to compensate for `jq` not having an `--inplace` flag. This will likely not
+   exist on your system but can normally be installed with the package manager
+   of your choice.
+
+   ```bash
+   PROVIDER_NAME=my-other-provider
+   jq '.Statement[].Condition."ForAnyValue:StringEquals"."${OIDC_PROVIDER_URL}:sub" |=
+       ( .+ ["system:serviceaccount:crossplane:'$(
+           kubectl -n crossplane get ControlLerConfig ${PROVIDERNAME} -o \
+               jsonpath='{.spec.serviceAccountName}'
+       )'"] | unique)' policies/crossplane-assume-role-trust-policy.json \
+           | sed '/".*:",/d' | sponge policies/crossplane-assume-role-trust-policy.json
+   ```
+
+1. Next, create a new WebIdentity role called `crossplane-assume-role` and bind
+   this policy to it. The Open ID Connect Identity Provider should be set to the
+   one for your cluster.
+
+   - substitute the variables `AWS_ACCOUNT_ID` and `OICD_PROVIDER_URL` in the
+     trust policy:
+
+     ```bash
+      export OIDC_PROVIDER_URL=$(aws iam list-open-id-connect-providers \
+          | jq '.OpenIDConnectProviderList[] | .Arn | split("/")[1] | select(. | test("^\\w+.mycluster.(?!mycluster)"))')
+      export AWS_ACCOUNT_ID=$(aws sts get-caller-identity | jq -rn .Account)
+      cat policies/crossplane-assume-role-trust-policy.json | envsubst \
+          | sponge policies/crossplane-assume-role-trust-policy.json
+     ```
+
+   - Check that the role does not already exist
+
+     ```bash
+     aws iam list-roles | jq '.Roles[] | select( .RoleName == "crossplane-assume-role")'
+     ```
+
+     If the role does not exist, we need to create it and attach the policy
+
+     ```bash
+       aws iam create-role --role-name crossplane-assume-role --policy-document file://policies/crossplane-assume-role-trust-policy.json
+       aws iam attach-role-policy --policy-arn \
+           "arn:aws:iam::aws:policy/crossplane-assume-role-policy" \
+           --role-name crossplane-assume-role
+     ```
+
+     If the role already exists, update its trust policy with
+
+     ```bash
+       aws iam update-assume-role-policy --role-name crossplane-assume-role \
+           --policy-document file://policies/crossplane-assume-role-trust-policy.json
```

Comment:
is this action additive? or you need to add existing permissions to `crossplane-assume-role-trust-policy.json` first?